[12.0][MIG] hr_period

AdriaGForgeFlow opened the pull request:
Migration of module hr_period to v12

Commit:
Module hr_period

saltonmassally committed:
Migrate hr_period from 7.0 to 8.0

serpentcs-dev1 committed:
hr_period v9

AaronHForgeFlow committed:
Adapt hr_period to date_range module

nikul-serpentcs committed:
[MIG] hr_period v10

AaronHForgeFlow committed:
[UPT]Adapt to date range model

nikul-serpentcs committed:
[IMP] Improved code

AaronHForgeFlow committed:
[UPT]improve test coverage

AaronHForgeFlow committed:
[FIX]on change hr period in payslips

AaronHForgeFlow committed:
[FIX]default period on payslips

oca-travis committed:
[UPD] Update hr_period.pot

MiquelRForgeFlow committed:
[FIX] hr_period: don't fail test in onchange_contract

oca-travis committed:
[UPD] Update hr_period.pot

gurneyalex committed:
Translated using Weblate (French)

Currently translated at 100.0% (82 of 82 strings)

Translation: hr-10.0/hr-10.0-hr_period
Translate-URL: https://translation.odoo-community.org/projects/hr-10-0/hr-10-0-hr_period/fr/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

AdriaGForgeFlow committed:
[12.0][MIG] Migrate module hr_period to v12

pedrobaeza commented:
Why this is called `hr_period` instead of `hr_payroll_period` if the use is for payrolls?

AdriaGForgeFlow commented:
According to @aheficent it should stay as hr_period.
https://github.com/OCA/hr/pull/368#issuecomment-333769013

pedrobaeza commented:
Then you should modify the README for expressing that.

AaronHForgeFlow committed:
[FIX]hr_period multicompany

nikul-serpentcs commented on hr_period/data/ir_sequence_data.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
@ageficent here `<odoo  noupdate="1">`

nikul-serpentcs commented on hr_period/tests/test_hr_fiscalyear.py:
```diff
@@ -0,0 +1,318 @@
+        fy.create_periods()
+        periods = self.get_periods(fy)
+        self.assertEqual(len(periods), 12)
+        self.assertEqual(periods[0].date_payment.strftime('%Y-%m-%d'),
```
@ageficent Here Use server `Date`, EveryWhere

nikul-serpentcs reviewed (changes requested):
Improve Code

oca-travis committed:
[UPD] Update hr_period.pot

Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

AaronHForgeFlow commented:
![youdontsay](https://user-images.githubusercontent.com/19620251/50004347-2e86af00-ffa7-11e8-802e-a787eccb0dc2.png)

pedrobaeza commented:
Have you attended all the comments?

naglis commented on hr_period/security/hr_period_security.xml:
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
```
File not included in `__manifest__.py`?

naglis commented on hr_period/views/date_range_type_view.xml:
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
File not included in `__manifest__.py`?

naglis commented on hr_period/views/hr_period_view.xml:
```diff
@@ -0,0 +1,87 @@
+        <field name="name">hr.period.tree</field>
+        <field name="model">hr.period</field>
+        <field name="arch" type="xml">
+          <tree colors="blue:state == 'open';gray:state == 'done'" string="Payroll Period">
```
OCA guideline: The `string` attribute in list views is deprecated (no longer displayed) since version 8.0 [[source](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#views)].

Same problem in:

- `views/hr_fiscalyear_view.xml`;
- `views/date_range_type_view.xml`.

---

OCA guideline: The `colors` attribute in list views is deprecated in favor of `decoration-{$name}` since version 9.0 [[source](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#views)].

Same problem in:
- `views/hr_fiscalyear_view.xml`.

naglis commented on hr_period/views/hr_fiscalyear_view.xml:
```diff
@@ -0,0 +1,95 @@
+        <field name="arch" type="xml">
+            <form string="Fiscal Year">
+                <header>
+                    <button name="create_periods" string="Create Periods" type="object" class="oe_highlight" states="draft"/>
```
`oe_highlight` no longer has CSS rules associated with it, it is being replaced by `btn-primary` since version 12.0 [[source](https://github.com/odoo/odoo/blob/1e5fbb8e5bf0e0458d83a399b2b59d03a601e86a/addons/web/static/src/js/core/dom.js#L340-L345)].

Same problem in:

- below in this file;
- `views/hr_payslip_run_view.xml`.

alexey-pelykh commented on oca_dependencies.txt:
```diff
@@ -1 +1,2 @@
 partner-contact
+server-ux https://github.com/OCA/server-ux
```
just `server-ux` should be enough?

alexey-pelykh commented on hr_period/tests/test_hr_fiscalyear.py:
```diff
@@ -282,10 +293,14 @@ def test_create_periods_monthly_payment_last_day(self):
-        self.assertEqual(periods[11].date_payment, '2015-12-31')
+        self.assertEqual(periods[0].date_payment.strftime('%Y-%m-%d'),
+                         '2015-01-31')
+        self.assertEqual(periods[1].date_payment.strftime('%Y-%m-%d'),
```
Maybe redo this to comparing to `date(2015, 2, 28)`?

MiquelRForgeFlow replied:
why?

alexey-pelykh replied:
less copy-paste of `strftime` + format?

MiquelRForgeFlow replied:
done

alexey-pelykh commented on hr_period/views/hr_fiscalyear_view.xml:
```diff
@@ -69,11 +69,11 @@
                 <field name="payment_day"/>
-                <filter string="Open" domain="[('state', '=', 'open')]" icon="terp-camera_test"/>
-                <filter string="Closed" domain="[('state', '=', 'done')]" icon="terp-dialog-close"/>
+                <filter name="Open" domain="[('state', '=', 'open')]" icon="terp-camera_test"/>
```
`name="open" string="Open"`?

alexey-pelykh commented on hr_period/views/hr_period_view.xml:
```diff
@@ -43,7 +43,7 @@
         <field name="model">hr.period</field>
         <field name="arch" type="xml">
-          <tree colors="blue:state == 'open';gray:state == 'done'" string="Payroll Period">
+          <tree decoration-info="state == 'open'" decoration-muted="state == 'done'" name="Payroll Period">
```
Why name instead of string?

MiquelRForgeFlow replied:
OCA guideline: The string attribute in list views is deprecated (no longer displayed) since version 8.0 [[source](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#views)].

alexey-pelykh replied:
That is so odd, there are very many similar usages in Odoo sources itself, while name is being used very rarely:
![image](https://user-images.githubusercontent.com/345102/62710771-8ffd0180-ba00-11e9-8126-3addab5aa5b1.png)

@pedrobaeza maybe you know why is that so?

alexey-pelykh replied:
To be exact, from https://www.odoo.com/documentation/12.0/reference/views.html
![image](https://user-images.githubusercontent.com/345102/62712168-ff73f080-ba02-11e9-9688-3e9436f37916.png)
but no `name` mentioned as well.

alexey-pelykh replied:
It seems that it's just ignored and can be simply removed 🤷‍♂

alexey-pelykh replied:
So, `string` is not shown, neither does `name`. I don't like spaces in names as that is technical key, but that's not blocking :) Since I've approved the PR already, this one is just FYI :P

alexey-pelykh commented:
Approving even with name/string question, as I think it's no harm in that

alexey-pelykh commented:
@nikul-serpentcs @naglis please update your reviews

MiquelRForgeFlow reviewed (approved):
=)

alexey-pelykh commented on hr_period/models/hr_fiscal_year.py:
```diff
@@ -0,0 +1,299 @@
+        return datetime(today.year, 12, 31).strftime(DF)
+
+    @api.model
+    def _default_type(self, company_id=False):
```
@mreficent would it be possible to test-cover this method?

alexey-pelykh replied:
Odd, it's actually called from https://github.com/OCA/hr/pull/524/files#diff-a191dafb7bdefc26785687b9040c0715R246

alexey-pelykh commented:
/ocabot merge
